### PR TITLE
 pybind/ceph_argparse.py:'timeout' must in kwargs when call run_in_thread

### PR DESCRIPTION
**Pull request by yangDL**
fix bug: Function "run_in_thread" get "timeout" from kwargs， but other functions passing "timeout" in args, so "timeout" is 0 forever in run_in_thread.

### CODE REVIEW

**Comment by tchaikov** [comment]
please see the definition of `Rados.osd_command()`, `Rados.pg_command()` and `Rados.mon_command()`. they do accept the `timeout` parameter.

**Comment by tchaikov** [comment]
gotcha! 
could you add a "`Signed-off-by`" line at the end of your commit message? "git commit -s" will do the trick for you. see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#1-sign-your-work and  summarize your change in the title of your commit? as it now reads like a bug report.

**Comment by yuriw** [comment]
test this please

**Comment by yuriw** [comment]
[wip-yuri-testing-2018-04-26-1557](https://trello.com/c/mkjV8uA7/300-wip-yuri-testing-2018-04-26-1557)

**Comment by yuriw** [comment]
looks like this PR caused failures in http://pulpito.ceph.com/yuriw-2018-04-26_20:34:49-rados-wip-yuri-testing-2018-04-26-1557-distro-basic-smithi/
dropping for now

**Comment by tchaikov** [comment]
```
2018-04-27T00:32:27.847 INFO:teuthology.orchestra.run.smithi161:Running: "sudo ceph --cluster ceph tell 'mon.*' injectargs -- --no-mon-health-to-clog"
2018-04-27T00:32:27.958 INFO:teuthology.orchestra.run.smithi161.stderr:2018-04-27 00:32:27.939 7f0d3bd76700 -1 WARNING: all dangerous and experimental features are enabled.
2018-04-27T00:32:27.967 INFO:teuthology.orchestra.run.smithi161.stderr:2018-04-27 00:32:27.951 7f0d3bd76700 -1 WARNING: all dangerous and experimental features are enabled.
2018-04-27T00:32:27.980 INFO:teuthology.orchestra.run.smithi161.stderr:Traceback (most recent call last):
2018-04-27T00:32:27.981 INFO:teuthology.orchestra.run.smithi161.stderr:  File "/usr/bin/ceph", line 1208, in <module>
2018-04-27T00:32:27.981 INFO:teuthology.orchestra.run.smithi161.stderr:    retval = main()
2018-04-27T00:32:27.981 INFO:teuthology.orchestra.run.smithi161.stderr:  File "/usr/bin/ceph", line 1124, in main
2018-04-27T00:32:27.981 INFO:teuthology.orchestra.run.smithi161.stderr:    prefix='get_command_descriptions')
2018-04-27T00:32:27.981 INFO:teuthology.orchestra.run.smithi161.stderr:  File "/usr/lib/python2.7/dist-packages/ceph_argparse.py", line 1364, in json_command
2018-04-27T00:32:27.981 INFO:teuthology.orchestra.run.smithi161.stderr:    raise RuntimeError('"{0}": exception {1}'.format(argdict, e))
2018-04-27T00:32:27.981 INFO:teuthology.orchestra.run.smithi161.stderr:RuntimeError: "None": exception "['{"prefix": "get_command_descriptions"}']": exception run_in_thread() got multiple values for keyword argument 'target'
2018-04-27T00:32:27.990 ERROR:teuthology.contextutil:Saw exception from nested tasks
```

File: src/pybind/ceph_argparse.py [hunk 1]
```diff
@@ -1281,18 +1281,18 @@ def send_command(cluster, target=('mon', ''), cmd=None, inbuf=b'', timeout=0,
...
                 print('{0} to {1}'.format(cmd, target[0]),
                       file=sys.stderr)
             if len(target) < 2 or target[1] == '':
                 ret, outbuf, outs = run_in_thread(
-                    cluster.mon_command, cmd, inbuf, timeout)
+                    cluster.mon_command, cmd, inbuf, timeout=timeout)
             else:
                 ret, outbuf, outs = run_in_thread(
-                    cluster.mon_command, cmd, inbuf, timeout, target[1])
+                    cluster.mon_command, cmd, inbuf, timeout=timeout, target=target[1])
```
Comment (tchaikov) [inline src/pybind/ceph_argparse.py]: please note, the first argument is the "target" which we need to run in a thread, while the last argument specifies the monitor on which we need to run this command.

File: src/pybind/ceph_argparse.py [hunk 2]
```diff
@@ -1172,16 +1172,16 @@ def find_cmd_target(childargs):
 
 
 class RadosThread(threading.Thread):
-    def __init__(self, target, *args, **kwargs):
+    def __init__(self, cluster_command, *args, **kwargs):
```
Comment (tchaikov) [inline src/pybind/ceph_argparse.py]: s/cluster_command/func/

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2018-04-30_00:59:17-rados-wip-kefu-testing-2018-04-29-1248-distro-basic-smithi/

**Pull request merged**